matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TimelineEventDecryptor.kt

```diff
@@ -72,16 +71,15 @@ internal class TimelineEventDecryptor(
 
     fun requestDecryption(eventId: String) {
         synchronized(existingRequests) {
-            if (existingRequests.contains(eventId)) {
-                return Unit.also {
-                    Timber.d("Skip Decryption request for event $eventId, already requested")
-                }
+            if (eventId in existingRequests) {
+                Timber.d("Skip Decryption request for event $eventId, already requested")
+                return
             }
             existingRequests.add(eventId)
         }
         synchronized(unknownSessionsFailure) {
             unknownSessionsFailure.values.forEach {
-                if (it.contains(eventId)) return@synchronized Unit.also {
+                if (eventId in it) {
```

Comment:
Oops, I missed the scope of that. Will fix.

Reply:
Ok, thanks. After analysis, it should be a return, but I will fix that myself after merging your PR.
Also we have discover that the thread can be block because there are nested `synchronized(unknownSessionsFailure)`... I will fix that also